### PR TITLE
Angular tidyup

### PR DESCRIPTION
**Pull request by Matt-Yorkley**
#### What? Why?

There's some nice things we could potentially add for UX on the new checkout, but we need to start cleaning up some old bits of Angular mess and code that causes Javascript memory leaks...

This is a bit of prep, including: deleting a few bits of dead code, avoiding attaching the `Darkswarm` module to `window`, and switching some bits of syntax that bind click events in Angular directives.

#### What should we test?

A green build should be enough here. If these things were not working we would definitely see some broken specs.

#### Release notes

Changelog Category: Technical changes

### CODE REVIEW

File: app/assets/javascripts/darkswarm/controllers/offcanvas_controller.js.coffee [hunk 1]
```diff
@@ -5,7 +5,7 @@ angular.module('Darkswarm').controller "OffcanvasCtrl", ($scope) ->
     $scope.menu.height($(window).height())
 
   $scope.bind = ->
-    $(window).on("resize", $scope.setOffcanvasMenuHeight)
+    $(window).bind("resize", $scope.setOffcanvasMenuHeight)
```
Comment (jibees) [inline app/assets/javascripts/darkswarm/controllers/offcanvas_controller.js.coffee:8]: jQuery still used here no?

Comment (Matt-Yorkley) [reply app/assets/javascripts/darkswarm/controllers/offcanvas_controller.js.coffee:8]: Ah you're right! I thought Angular had it's own `bind()` function, but I think it's a bit different. I'll re-assess this bit :+1:

Comment (Matt-Yorkley) [reply app/assets/javascripts/darkswarm/controllers/offcanvas_controller.js.coffee]: Okay I dug a bit deeper and it looks like Angular includes a min version of `jQuery` anyway, called `jqLite`. If `jQuery` is present, Angular will use that, but if it's not it uses it's own `jqLite` implementation which includes methods like `bind()`.

Anyway, I've switched strategy a bit here, and added some explicit teardown in directives that bind events on DOM elements: https://github.com/openfoodfoundation/openfoodnetwork/pull/8005/commits/15b6b1271a92a393088dd35a5a5628b37ff14d00

**Review by jibees — APPROVED** [review]
👌 🙏

**Pull request merged**